### PR TITLE
Securelaunch cleanup/fix-ups

### PR DESCRIPTION
**Pull request by pjcolp**
Cleanup comments in the securelaunch code, moving from "/* ... */" style comments to "// ..." style comments. Remove some dead code and fix a couple of small bugs.

### CODE REVIEW

File: pkg/securelaunch/helpers.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func WriteToFile(data []byte, dst, defFileName string) (string, error) {
 // persist writes data to targetPath.
 // targetPath is of form sda:/boot/cpuid.txt
 func persist(data []byte, targetPath string, defaultFile string) error {
-	filePath, r := GetMountedFilePath(targetPath, 0) // 0 is flag for rw mount option
+	filePath, r := GetMountedFilePath(targetPath, mount.MS_NOEXEC) // MS_NOEXEC means RW
```
[the remark this answers is not in the document]
Comment (pjcolp) [reply pkg/securelaunch/helpers.go:78]: Hmmm. I know I hit a bug with this before and that seemed to be the issue/resolution (that was over a year ago, so maybe things have changed). But playing around with it now, I don't see any difference in behaviour one way or the other. I also don't see any MS_* flag that maps to 0 either, though. I could edit the commit message to say it's just changing it to an actually defined flag or just drop the commit.

**Comment by pjcolp** [comment]
Posting here as well, in case posting a reply to the comment on the code file doesn't trigger a notification.

Hmmm. I know I hit a bug with this before and that seemed to be the issue/resolution (that was over a year ago, so maybe things have changed). But playing around with it now, I don't see any difference in behaviour one way or the other. I also don't see any MS_* flag that maps to 0 either, though. I could edit the commit message to say it's just changing it to an actually defined flag or just drop the commit.

File: pkg/securelaunch/helpers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/securelaunch/helpers.go:78]: I would incline toward dropping the commit *unless* you want NOEXEC behavior, i.e. you can't run programs on the mounted file system?

**Review by rminnich — COMMENTED** [review]
save for my confusion on the MS_NOEXEC commit, this looks good to go, can you take another look at that and see what you want to do?

**Comment by rminnich** [comment]
OK, this has nothing to do with the mount, the directory is created by pkg/mount, and it's a mistake, we let it slip through in 74c94c3e4e7b8a9996fba608ca0124b65b21ffcb

That commit needs to be reverted, I don't want to guess how much stuff it will break, but it's just wrong.

So, yeah, delete your MS_NOEXEC mount commit, it's not doing what it seems.

File: pkg/securelaunch/helpers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/securelaunch/helpers.go:78]: I did just verify that the u-root mount command is creating the mountpoint: 
[pid 397] E mkdirat(0xffffffffffffff9c, 0xc0001f44d0 /bbb, 0x1b6)
So this is u-root, not the kernel, and I"m turning that off, this is not good behavior. 

Your commit is not necessary, and if you want the directory to be pre-made, there's now a way to do
so in the u-root package. But this commit is not needed in its form.

See #2395 for details on how to get the directory made if needed.

**Review by rminnich — CHANGES_REQUESTED** [review]

**Comment by pjcolp** [comment]
I've removed the commit that changes the mount flag and rebased on top of the current main branch.

**Review by rminnich — APPROVED** [review]

**Comment by rminnich** [comment]
Thanks!

**Comment by rminnich** [comment]
Note the fix went in to no longer automagically create mount points, I am glad you pointed this out to me.

**Pull request merged**